hrl/agent/dsc/dsc.py

```diff
@@ -6,41 +6,37 @@
 import numpy as np
 
 from hrl.agent.dsc.utils import *
-from hrl.agent.dsc.MBOptionClass import ModelBasedOption
+from hrl.option import ModelBasedOption
 
 
-class RobustDSC(object):
-    def __init__(self, mdp, warmup_episodes, max_steps, gestation_period, buffer_length, use_vf, use_global_vf, use_model,
-                 use_diverse_starts, use_dense_rewards, lr_c, lr_a, clear_option_buffers,
-                 use_global_option_subgoals, maze_type, experiment_name, device,
-                 logging_freq, generate_init_gif, evaluation_freq, seed, multithread_mpc):
+class RobustDSC:
+    def __init__(self, mdp, params):
```

Comment:
I left it this way because I thought that too many arguments was bad for style. But you're right, the DSC class should be self-contained, and users need to know what the input params are. 

I can either revert back to the multiple-argument version, or create documentation on what the `params` dictionary needs to looks like. Probably will do the former so it's easier to avoid bad inputs.

Reply:
have reverted back to the multiple-argument version on another branch!

Reply:
and also on this branch